### PR TITLE
samples: Add examples for making requests in a salted row key scenario

### PR DESCRIPTION
**Pull request by billyjacobson**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigtable/KeySalting.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      rows.addAll(future.get());
+    }
+
+    System.out.printf("Successfully fetched %s rows\n", rows.size());
+    for (Row row : rows) {
+      System.out.printf("Successfully read row %s\n", row.getKey().toStringUtf8());
+    }
+  }
+
+  public static String getSaltedRowKey(String rowKey, int saltRange) {
```
Comment (kolea2) [inline samples/snippets/src/main/java/com/example/bigtable/KeySalting.java:80]: can you add some lightweight javadoc here on what a salted row key is and what this is doing?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: samples/snippets/src/main/java/com/example/bigtable/KeySalting.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId);
+
+    List<Query> queries = new ArrayList<>();
+    for (int i = 0; i < SALT_RANGE; i++) {
+      queries.add(Query.create(tableId).prefix(i + "-" + prefix));
+    }
+
+    List<ApiFuture<List<Row>>> futures = new ArrayList<>();
+    for (Query q : queries) {
+      futures.add(dataClient.readRowsCallable().all().futureCall(q));
```
Comment (kolea2) [inline samples/snippets/src/main/java/com/example/bigtable/KeySalting.java:66]: why not just use `client.readRowsAsync(...`?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: samples/snippets/src/main/java/com/example/bigtable/KeySalting.java (hunk 1 above)
Comment (billyjacobson) [reply samples/snippets/src/main/java/com/example/bigtable/KeySalting.java:91]: Added, let me know if it's clear

File: samples/snippets/src/main/java/com/example/bigtable/KeySalting.java (hunk 2 above)
Comment (billyjacobson) [reply samples/snippets/src/main/java/com/example/bigtable/KeySalting.java:66]: Let me know what Igor says about this. I'm happy to change it to something more clear, but I don't know how to use the clients for async stuff.

**Pull request merged**